akka-persistence-typed/src/main/scala/akka/persistence/typed/javadsl/ReplicatedEventSourcing.scala

```diff
@@ -122,4 +148,29 @@ object ReplicatedEventSourcing {
     eventSourcedBehaviorFactory(context)
   }
 
+  /**
+   * Initialize a replicated event sourced behavior where each journal has different journal configuration e.g.
+   * each replica uses a different database or requires different database configuration for a shared database.
+   *
+   * Events from each replica for the same entityId will be replicated to every copy.
+   * Care must be taken to handle events in any order as events can happen concurrently at different replicas.
+   *
+   * Using an replicated event sourced behavior means there is no longer the single writer guarantee.
+   *
+   * The journal plugin id for the entity itself can be configured using withJournalPluginId after creation.
+   * A query side identifier is passed per replica allowing for separate database/journal configuration per
+   * replica. The events from other replicas are read using PersistentQuery.
+   *
+   * @param allReplicasAndQueryPlugins All replica ids and a query plugin per replica id. These need to be known to receive events from all replicas
+   *                                   and configured with the query plugin for the journal that each replica uses.
+   */
+  def onCommandPerReplicaJournalConfig[Command, Event, State](
```

Comment:
Mouthful, but could be good with same prefix:
`perReplicaJournalConfigForEventSourcedOnCommandBehavior`

similar with commonJournalConfig